Setup Players, Game and Message classes

timonrey opened the pull request:
The code includes a new game class, where the game can be activated. 
The buttonWasPressedWithValue method will be triggered by the button
In the Main class, we have new instances of the Players, Game and Message class.
The methods in the new Message class can tell you the information of the button push.
The Players class is going to change the score of each player after a button push.

timonrey committed:
newCodeForScore

PhilippSpo commented:
> The Players class is going to change the score of each player after a button push.

The Players class will not change any value. What will instead?

PhilippSpo commented on src/Game.java:
```diff
@@ -0,0 +1,35 @@
+
+    Players playerOne;
+    Players playerTwo;
+    Players scorePlayerOne = new Players();
```
This is still the old code. I guess you have forgot to commit 😉

emmenko replied:
Should I wait before reviewing this?

TimonRey commented:
Can i make a new pull request? i have changed and deleted some names and comments…? 
Would that make sense? 



Timon Turak
Trainee

E. timon.turak@commercetools.de

commercetools GmbH
Adams-Lehmann-Str. 44
80797 München

Amtsgericht München, HRB 161496
Geschäftsführer: Denis Werner, Dirk Hörig, Udo Rauch, Dr. Robert Zores

> Am 28.11.2017 um 14:28 schrieb Philipp Sporrer <notifications@github.com>:
> 
> @PhilippSpo requested changes on this pull request.
> 
> In src/Game.java <https://github.com/TimonRey/PingPongApp/pull/13#discussion_r153490566>:
> 
> > @@ -0,0 +1,35 @@
> +public class Game {
> +
> +    Players playerOne;
> +    Players playerTwo;
> +    Players scorePlayerOne = new Players();
> This is still the old code. I guess you have forgot to commit 😉
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/TimonRey/PingPongApp/pull/13#pullrequestreview-79502267>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AeS7SQbGQRgGd3HiYIKU4ZEOqy18xGunks5s7ApogaJpZM4Qs9OD>.
> 


-- 


[image: commercetools logo] <http://www.commercetools.com>


Adams-Lehmann-Straße 44, 80797 Munich | Offices: Sonnenallee 223, 12059 
Berlin

commercetools GmbH, Registered Office: Munich, Register: HRB 161496 at the 
Local Court (Amtsgericht) of Munich

Members of the Board of Management: Dirk Hörig | Denis Werner | Udo Rauch | 
Dr. Robert Zores

 

commercetools Inc. 318 Blackwell St. Suite 240, Durham, NC 27701

Members of the Board of Management: Dirk Hörig | Arthur Lawida


<http://go.commercetools.com/api-booklet?utm_source=Mail_EN&utm_medium=Mail_Footer_CT&utm_term=API_Booklet&utm_content=API_Booklet&utm_campaign=APIBooklet>

PhilippSpo commented:
Just commit the stuff on this branch to update the PR

TimonRey commented:
Instead of the Player class, the methods in this class will change the score value of the player who pushed the button



Timon Turak
Trainee

E. timon.turak@commercetools.de

commercetools GmbH
Adams-Lehmann-Str. 44
80797 München

Amtsgericht München, HRB 161496
Geschäftsführer: Denis Werner, Dirk Hörig, Udo Rauch, Dr. Robert Zores

> Am 28.11.2017 um 14:27 schrieb Philipp Sporrer <notifications@github.com>:
> 
> The Players class is going to change the score of each player after a button push.
> 
> The Players class will not change any value. What will instead?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/TimonRey/PingPongApp/pull/13#issuecomment-347522944>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AeS7Sf7CFnPbBrBSS1gMspXfUj3Obn8eks5s7ApAgaJpZM4Qs9OD>.
> 


-- 


[image: commercetools logo] <http://www.commercetools.com>


Adams-Lehmann-Straße 44, 80797 Munich | Offices: Sonnenallee 223, 12059 
Berlin

commercetools GmbH, Registered Office: Munich, Register: HRB 161496 at the 
Local Court (Amtsgericht) of Munich

Members of the Board of Management: Dirk Hörig | Denis Werner | Udo Rauch | 
Dr. Robert Zores

 

commercetools Inc. 318 Blackwell St. Suite 240, Durham, NC 27701

Members of the Board of Management: Dirk Hörig | Arthur Lawida


<http://go.commercetools.com/api-booklet?utm_source=Mail_EN&utm_medium=Mail_Footer_CT&utm_term=API_Booklet&utm_content=API_Booklet&utm_campaign=APIBooklet>

emmenko commented:
@TimonRey don't answer from the Email, come here in Github and answer in the website. Thanks ;)

TimonRey commented:
@PhilippSpo Instead of the Player class, the methods in this class will change the score value of the player who pushed the button

timonrey committed:
minor improvements

timonrey committed:
Merge branch 'master' into setup-players-game-classes

timonrey committed:
remove unused playerTwo class

emmenko commented on src/Game.java:
```diff
@@ -0,0 +1,32 @@
+    Players playerOne;
+    Players playerTwo;
+
+    public Game(Players firstPlayerName, Players secondPlayerName) {
```
Why is it called `xxxName`?

timonrey replied:
It makes it easier for me to understand what the __constructor__ will *accept*

PhilippSpo replied:
But the constructor won't accept player **names**. It accepts the whole **player** itself, not only the name.

emmenko commented on src/Game.java:
```diff
@@ -0,0 +1,32 @@
+        
+    }
+
+    public void buttonWasPressedWithValue(Message mail) {
```
Nitpick: we should probably rename this method. Also why is the variable called `mail`? 😄

timonrey replied:
You're right, the name is too long. I don't know why but the name before was more ___stupid___ :)

emmenko commented on src/Main.java:
```diff
@@ -0,0 +1,79 @@
+
+       // match.playerTwo = new Players()
+
+        match.buttonWasPressedWithValue(nicolaScores);
```
Wouldn't be better to create a new message each time you execute this function?

`match.buttonWasPressedWithValue(new Message(...))`

timonrey replied:
That's included in the ___new version___, i think

timonrey committed:
Colocate classes

emmenko commented on src/Message.java:
```diff
@@ -0,0 +1,21 @@
+    public Message (Players playerNumber, int actionType) {
+        this.player = playerNumber;
+        this.actionType = actionType;
+    }
```
The `Message` should look like this right?

https://github.com/TimonRey/PingPongApp/blob/89ee8306d258e9b731add35f5c784c8258fe3749/src/main/java/PingPong.java#L48-L51

So what you get from a `Message` is
* `buttonId`: it's a string
* `actionType`: also a string

timonrey replied:
No, the __button__ variable is a *string* but the __actionType__ is an *int*...just a number

emmenko commented on src/Players.java:
```diff
@@ -0,0 +1,27 @@
+    }
+
+
+    public int getter () {
```
Maybe we can rename this to something like `getScore` or `getPoints`, `getter` is a bit generic

timonrey replied:
# Sure!
 it was just practically for me to __understand__ what a getter method is

emmenko commented on src/Players.java:
```diff
@@ -0,0 +1,27 @@
+    }
+
+    public void sub () {
+        points = points - 1;
```
Nitpick: you should probably don't allow negative numbers

timonrey replied:
oh...you're right! But the __sub method__ will be changed into a __set method__ in the next step

emmenko reviewed (dismissed):
Good work so far! 🙌

I left some initial comments, mostly related to naming ;)

timonrey committed:
Setup players with button data

timonrey committed:
Renaming internal methods
